compiler/noirc_frontend/src/ast/mod.rs

```diff
@@ -390,15 +390,16 @@ pub enum Visibility {
     Private,
     /// DataBus is public input handled as private input. We use the fact that return values are properly computed by the program to avoid having them as public inputs
     /// it is useful for recursion and is handled by the proving system.
-    DataBus,
+    /// The u32 value is used to group inputs having the same value.
+    DataBus(u32),
 }
 
 impl std::fmt::Display for Visibility {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
             Self::Public => write!(f, "pub"),
             Self::Private => write!(f, "priv"),
-            Self::DataBus => write!(f, "databus"),
+            Self::DataBus(_) => write!(f, "databus"),
```

Comment:
It seems useful to include the databus ID?
```suggestion
            Self::DataBus(id) => write!(f, "databus({})", id),
```

Reply:
Done